index.ts

```diff
@@ -0,0 +1,44 @@
+interface ITreeNode<TValue> {
+  [x: string]: any;
+  left?: ITreeNode<TValue>
+  right?: ITreeNode<TValue>
+  value: TValue
+}
+
+class Tree<TValue extends number> {
+  public root?: ITreeNode<TValue>;
+
+  public insert(value: TValue) {
+    if (this.root) {
+      this._insert(this.root, value)
+    } else {
+      this.root = {
+        value,
+      }
+    }
+  }
+
+  private _insert(node: ITreeNode<TValue>, value: TValue) {
+    let direction = 'left'
+    if (value >= node.value) {
+      direction = 'right'
+    }
```

Comment:
You can make it look better

```suggestion
const direction = value >= node.value ? 'right' : 'left'
```